gtsfm/loader/astronet_loader.py

```diff
@@ -0,0 +1,231 @@
+from thirdparty.colmap.scripts.python.read_write_model import Point3D as ColmapPoint3D
+
+
+logger = logger_utils.get_logger()
+
+
+class AstroNetLoader(LoaderBase):
+    """Simple loader class that reads an AstroNet data segment.
+
+    Refs:
+    - https://github.com/travisdriver/astronet
+    """
+
+    def __init__(
+        self,
+        data_dir: str,
+        use_gt_extrinsics: bool = True,
+        use_gt_sfmtracks: bool = False,
+        max_frame_lookahead: int = 2,
+    ) -> None:
+        """Initialize loader from a specified segment directory (data_dir) on disk.
+
+        <data_dir>/
+             ├── images/: undistorted grayscale images
+             ├── cameras.bin: camera calibrations (see https://colmap.github.io/format.html#cameras-txt)
+             ├── images.bin: 3D poses and 2D tracks (see https://colmap.github.io/format.html#images-txt)
+             └── points3D.bin: 3D tracks (see https://colmap.github.io/format.html#points3d-txt)
+
+
+        Args:
+            data_dir: path to directory containing the COLMAP-formatted data: cameras.bin, images.bin, and points3D.bin
+            use_gt_extrinsics (optional): whether to use ground truth extrinsics. Used only for comparison with
+                reconstructed values.
+            use_gt_sfmtracks (optional): whether to use ground truth tracks. Used only for comparison with reconstructed
+                values.
+            max_frame_lookahead (optional): maximum number of consecutive frames to consider for
+                matching/co-visibility. Any value of max_frame_lookahead less than the size of
+                the dataset assumes data is sequentially captured.
+
+        Raises:
+            FileNotFoundError if `data_dir` doesn't exist or image path does not exist.
+            RuntimeError if ground truth camera calibrations not provided.
+        """
+        self._use_gt_extrinsics = use_gt_extrinsics
+        self._use_gt_sfmtracks = use_gt_sfmtracks
+        self._max_frame_lookahead = max_frame_lookahead
+
+        # Use COLMAP model reader to load data and convert to GTSfM format.
+        if Path(data_dir).exists():
+            cameras, images, points3d = read_model(path=data_dir, ext=".bin")
+            self._calibrations, self._wTi_list, img_fnames, self._sfmtracks = self.colmap2gtsfm(
+                cameras, images, points3d, load_sfmtracks=use_gt_sfmtracks
+            )
+        else:
+            raise FileNotFoundError("No data found at %s." % data_dir)
+
+        # Camera intrinsics are currently required due to absence of EXIF data and diffculty in approximating focal
+        # length (usually 10000 to 100000 pixels).
+        if self._calibrations is None:
+            raise RuntimeError("Camera intrinsics cannot be None.")
+
+        if self._wTi_list is None:
+            self._use_gt_extrinsics = False
+
+        if self._sfmtracks is None:
+            self._use_gt_sfmtracks = False
```

Comment:
This is weird from a design perspective. if the user specifies use_gt_extrinsics or use_gt_sfmtracks, and they are not found, I think its better to throw an error than to silently force them to `False`. Maybe they had the wrong path or something.

Reply:
To do that without even logging a warning would be very user-unfriendly :)

Reply:
I agree; I was following convention of the other loaders. Should I change or keep the same as others?

Reply:
I think its better to raise exceptions if its None. The pipeline will crash, but that is probably the better action here, instead of waiting for the results only to realize we did not use ground truth. 
If its an easy change, could you do the same for the other loaders? Or maybe add a TODO? Asking for @johnwlambert 's or @ayushbaid 's opinion here.

Reply:
Maybe I could raise a warning instead? The extrinsics and sfmtracks aren't required to reconstruct the scene and are used strictly for computing metrics, so I could see a case for either.

Reply:
I think it depends on how general purpose we expect the loader to be.

For the Olsson loader, if no GT is found, we still proceed, since we assume it's ok for the user to just give us a folder of images without GT, since that's a valid situation.

For the Astronet data, if we assume GT is always readily available, then we should probably raise an error. But if downloading the GT might be cumbersome, and someone just wants to try, then maybe no error, but a helpful warning message would be appreciated.

Reply:
Actually, I think raising an exception makes more sense, because if we just ignore that the inputs are missing when they're requested by the user, it kind of defeats the purpose of the `_use_gt_intrinsics`, `_use_gt_extrinsics`, and `_use_gt_sfmtrakcs` variables.

Reply:
Added `RuntimeError` if ground truth requested but not found.